pointerevents/pointerevent_element_haspointercapture.html

```diff
@@ -0,0 +1,108 @@
+                                      "before target0.setPointerCapture, target0.hasPointerCapture should be false");
+                    });
+                    target1.setPointerCapture(e.pointerId);
+                    test_pointerEvent.step(function () {
+                        assert_equals(target0.hasPointerCapture(e.pointerId), false,
+                                      "target1.setPointerCapture, target0.hasPointerCapture should be false");
+                    });
+                    target0.setPointerCapture(e.pointerId);
+                    set_capture_to_target0 = true;
+                    // hasPointerCapture will return true immediately after a call to setPointerCapture
+                    test_pointerEvent.step(function () {
+                        assert_equals(target0.hasPointerCapture(e.pointerId), true,
+                                      "after target0.setPointerCapture, target0.hasPointerCapture should be true");
+                    });
+                    // hasPointerCapture will return false immediately after a call to releasePointerCapture
+                    target0.releasePointerCapture(e.pointerId);
+                    set_capture_to_target0 = false;
+                    test_pointerEvent.step(function () {
+                        assert_equals(target0.hasPointerCapture(e.pointerId), false,
+                                      "after target0.releasePointerCapture, target0.hasPointerCapture should be false");
+                    });
+                    target0.setPointerCapture(e.pointerId);
+                    set_capture_to_target0 = true;
+                    test_pointerEvent.step(function () {
+                        assert_equals(target0.hasPointerCapture(e.pointerId), true,
+                                      "after target0.setPointerCapture, target0.hasPointerCapture should be true");
+                    });
+                });
+
+                for (var i = 0; i < listening_events.length; i++) {
+                    on_event(target0, listening_events[i], function (e) {
+                        test_pointerEvent.step(function () {
+                            assert_equals(target0.hasPointerCapture(e.pointerId), set_capture_to_target0,
+                                          "Received " + e.type + " target0.hasPointerCapture should be " + set_capture_to_target0);
+                        });
+                    });
+                }
+
+                on_event(target0, "pointerup", function (e) {
+                    // Immediately after firing the pointerup or pointercancel events, a user agent must run the steps
+                    // as if the releasePointerCapture() method has been called
+                    test_pointerEvent.step(function () {
+                        assert_equals(target0.hasPointerCapture(e.pointerId), true,
+                                      "pointerup target0.hasPointerCapture should be true");
```

Comment:
We need to cover the case that pointerup gets a |false| for hasPointerCapture. What about adding a "pointerup" listener on target1 that expects a |false|, then add a 4th step saying "Click on yellow"?